codersdk/workspacebuilds.go

```diff
@@ -123,9 +123,29 @@ func (c *Client) WorkspaceBuild(ctx context.Context, id uuid.UUID) (WorkspaceBui
 	return workspaceBuild, json.NewDecoder(res.Body).Decode(&workspaceBuild)
 }
 
+type CancelWorkspaceBuildStatus string
+
+const (
+	CancelWorkspaceBuildStatusRunning CancelWorkspaceBuildStatus = "running"
+	CancelWorkspaceBuildStatusPending CancelWorkspaceBuildStatus = "pending"
+)
+
+type CancelWorkspaceBuildParams struct {
+	// ExpectStatus ensures the build is in the expected status before canceling.
+	ExpectStatus CancelWorkspaceBuildStatus `json:"expect_status,omitempty"`
```

Comment:
```suggestion
	ExpectStatus CancelWorkspaceBuildStatus `json:"-"`
```

Since it's never used as JSON

Reply:
Or just omit the `json` tag entirely perhaps